pandas/_libs/src/vendored/numpy/datetime/np_datetime.c

```diff
@@ -752,66 +752,56 @@ void pandas_timedelta_to_timedeltastruct(npy_timedelta td,
     }
 
     const npy_int64 per_day = sec_per_day * per_sec;
-    npy_int64 frac;
+    const int sign = td < 0 ? -1 : 1;
+    const int is_negative = td < 0 ? 1 : 0;
+    const int uneven_in_seconds = td % per_sec != 0 ? 1 : 0;
     // put frac in seconds
-    if (td < 0 && td % per_sec != 0)
-      frac = td / per_sec - 1;
-    else
-      frac = td / per_sec;
-
-    const int sign = frac < 0 ? -1 : 1;
-    if (frac < 0) {
+    npy_int64 sfrac = td / per_sec - is_negative * uneven_in_seconds;
```

Comment:
Am I reading this right that it just can add/subtract 1 when uneven_in_seconds is true? I might be missing the point but that _feels_ a bit off - not sure we have full test coverage in all precisions

Reply:
Thanks for scratching your head at this. It's a leftover from a different optimisation route that was not successful in the end. I moved the subtraction into the if body again (same as in the main version) so it should be clearer now. Also now the speedup is at where I had it with the previous version that did not pass the tests.